datastore/datastore/storagebackend/postgresql/getobservations.go

```diff
@@ -14,6 +16,103 @@ import (
+
+	obspb2go map[string]string // association between observation specific protobuf name and
+	// (generated by protoc) Go name
+	// NOTES:
+	// - Protobuf names are field names of the ObsMetadata message in datastore.proto.
+	// - Only fields of string type are included.
+	// - The observation value field is not included.
+	// - Protobuf names are identical to corresponding database column names.
+	// - Protobuf names are snake case (aaa_bbb) whereas Go names are camel case (aaaBbb or AaaBbb).
+
+	obsStringMdataGoNames []string // Go names for observation metadata of type string
+	obsStringMdataCols    []string // column names qualified with table name 'observation'
+
+	pb2table map[string]string // association between a protobuf name and a table name
+	// NOTES:
+	// - Protobuf names are field names of *either* the ObsMetadata *or* the TSMetadata message in
+	//   datastore.proto. As a consequence, the same field name may not occur in both messages.
+	// - Only fields of string type are included.
+	// - The observation value field is not included.
+	// - Protobuf names are identical to corresponding database column names.
+	// - Protobuf names are snake case (aaa_bbb).
+
+	pb2tableKeys []string // for error reporting
+)
+
+func init() {
+
+	// create tsStringMdataGoNames and tsStringMdataPBNames
+	tsStringMdataGoNames = []string{}
+	tsStringMdataPBNames = []string{}
+	for _, field := range reflect.VisibleFields(reflect.TypeOf(datastore.TSMetadata{})) {
+		if field.IsExported() && (field.Type.Kind() == reflect.String) {
+			goName := field.Name
+			tsStringMdataGoNames = append(tsStringMdataGoNames, goName)
+			tsStringMdataPBNames = append(tsStringMdataPBNames, common.ToSnakeCase(goName))
+		}
+	}
+
+	// create obspb2go, obsStringMdataGoNames, and obsStringMdataCols
+	obspb2go = map[string]string{}
+	obsStringMdataGoNames = []string{}
+	obsStringMdataCols = []string{}
+	for _, field := range reflect.VisibleFields(reflect.TypeOf(datastore.ObsMetadata{})) {
+		if field.IsExported() && (field.Type.Kind() == reflect.String) &&
+			(strings.ToLower(field.Name) != "value") { // (obs value not considered metadata here)
+			goName := field.Name
+			pbName := common.ToSnakeCase(goName)
+			obspb2go[pbName] = goName
+			obsStringMdataGoNames = append(obsStringMdataGoNames, goName)
+			obsStringMdataCols = append(obsStringMdataCols, fmt.Sprintf("observation.%s", pbName))
+		}
+	}
+
+	// create pb2table
+	pb2table = map[string]string{}
+	for _, pbName := range tsStringMdataPBNames {
+		pb2table[pbName] = "time_series"
+	}
+	for pbName := range obspb2go {
+		if _, found := pb2table[pbName]; found {
+			panic(fmt.Sprintf(
+				"precondition failed: field %s in ObsMetadata already found in TSMetadata", pbName))
+		}
+		pb2table[pbName] = "observation"
+	}
+
+	// create pb2tableKeys
+	pb2tableKeys = []string{}
+	for key := range pb2table {
+		pb2tableKeys = append(pb2tableKeys, key)
+	}
+	sort.Strings(pb2tableKeys)
+}
```

Comment:
Move this to a seperate file?

Reply:
That is a good idea. How about introducing an init.go (in the same directory) that takes care of all initialization for the postgresql package? Then we can move all package variables and their initialization code to that init.go. Note that there is no concept of file scope in Go, so all package variables (i.e. starting with lowercase and thus only accessible within the package) might as well be declared in a single file. And we could use the same organization for other packages too.

Reply:
Added [a commit](https://github.com/EURODEO/e-soh/pull/23/commits/bcf9a21719cfe879627192a43b7bc27ef4c9c8ed) to the PR just now for this.